WoWsShipBuilder.Web/Pages/Charts.razor

```
@@ -149,365 +150,310 @@
-                var shipSummary = AppData.ShipSummaryList!.Single(x => x.Index.Equals(shipIndex));
-                var ship = await WebAppDataService.GetShipFromSummary(shipSummary);
-                Dictionary<string, ArtilleryData> artilleryData = new();
-                if (ship!.MainBatteryModuleList != null && ship.MainBatteryModuleList.Count > 0)
-                {
-                    IEnumerable<string> shellIndexes = ship.MainBatteryModuleList.SelectMany(turretModule => turretModule.Value.Guns.SelectMany(gun => gun.AmmoList)).Distinct();
-                    foreach (string shellIndex in shellIndexes)
-                    {
-                        artilleryData.Add(shellIndex, new());
-                    }
-                }
-
-                ShellSelectionWrapper wrapper = new(shipIndex, artilleryData);
-                shipList.Add(wrapper);
-            }
-
-            DialogOptions options = new()
-            {
-                MaxWidth = MaxWidth.Small,
-                FullWidth = true,
-            };
-            var parameters = new DialogParameters
-            {
-                ["shipList"] = shipList,
-            };
-            var dialog = DialogService.Show<ShellSelectionDialog>("ShellSelectionDialog", parameters, options);
-            var result = await dialog.Result;
-            if (result is null || result.Cancelled)
-            {
-                processing = false;
-                return;
-            }
-            selectedShipList = ((IEnumerable<ShellSelectionWrapper>) result.Data).ToList();
+            StateHasChanged();
         }
-        await UpdateCharts(selectedShipList);
-        processing = false;
-        StateHasChanged();
     }
-
-    private string GetColor()
+        
+    async ValueTask IAsyncDisposable.DisposeAsync()
     {
-        if (colorCounter > ChartColors.Length - 1)
+        if (module is not null)
         {
-            colorCounter = 0;
+            await module.DisposeAsync();
```

Catch exception when disposing. If DisposeAsync is called after the browser window is closed, this will always throw an exception.
Do not catch all exceptions, only catch relevant exceptions!